pysat/instruments/cosmic_gps.py

```diff
@@ -113,47 +113,38 @@ def list_files(tag=None, sat_id=None, data_path=None, format_str=None):
     # overloading revision keyword below
     if format_str is None:
         # COSMIC file format string
-        if tag == 'scnlv1':
-            format_str = ''.join(('????.???/??????_C{revision:03d}.{year:04d}',
-                                  '.{day:03d}.{hour:02d}.{minute:02d}.',
-                                  '????.?{second:02d}.??_????.????_nc'))
-        else:
-            format_str = ''.join(('????.???/??????_C{revision:03d}.{year:04d}',
-                                  '.{day:03d}.{hour:02d}.{minute:02d}.',
-                                  '?{second:02d}_????.????_nc'))
-
+        format_str = ''.join(('*.*/*.{year:04d}.{day:03d}',
+                              '.{hour:02d}.{minute:02d}.*_nc'))
     # process format string to get string to search for
-    search_dict = pysat._files.construct_searchstring_from_format(format_str,
-                                                                  wildcard=False)
+    search_dict = pysat._files.construct_searchstring_from_format(format_str)
     search_str = search_dict['search_string']
     # perform local file search
     files = pysat._files.search_local_system_formatted_filename(data_path,
                                                                 search_str)
     # we have a list of files, now we need to extract the information
     # pull of data from the areas identified by format_str
-    stored = pysat._files.parse_fixed_width_filenames(files, format_str)
-
+    stored = pysat._files.parse_delimited_filenames(files, format_str,
+                                                    delimiter='.')
     if len(stored['year']) > 0:
         year = np.array(stored['year'])
         day = np.array(stored['day'])
         hour = np.array(stored['hour'])
         minute = np.array(stored['minute'])
```

Comment:
So. These are required to be `int`, but `_files.py` no longer requires all key arrays be integers.  Thus, we need a test here to handle the possibility that any one of these four values could be a string.

Reply:
Hmm.  We do need to add tests for cosmic, since it's outside the eye of pytest.  This does throw an error if I force `minute` as a string:
```
UFuncTypeError: ufunc 'multiply' did not contain a loop with signature matching types (dtype('<U32'), dtype('<U32')) -> dtype('<U32')
```
This should happen if the format string does not match the cosmic files.  We could specifically scan for these and produce a more helpful error message if any of the values are not integer.

Since this part of the code relies on an external list of files, I'm not really sure how to test for this without adding a lot of new infrastructure (say, a good list and a bad list of files stashed in .pysat).

Reply:
> Since this part of the code relies on an external list of files, I'm not really sure how to test for this without adding a lot of new infrastructure (say, a good list and a bad list of files stashed in .pysat).

We will need to develop that to properly unit test the file management.  I think that's outside of this PR, though.  I think that just adding the tests in this routine for now.  Later we can add unit test structure that provides Instruments with bad filenames and ensures the Instrument object handles the failure gracefully.

Reply:
Poking into this more, if `year` or `month` is passed through as a string, I do not see an error and the index creation line at 142 (new code) works as expected.  The error comes from the calculation of uts at line 133 (new), since we try to multiply a string by an int if `hour` or `minute` is passed through as a string.  I've added a possible testing path forward (for when things go right) as #443, which is likely outside the scope here.  I'm still not sure how to build a test for when things go wrong as suggested above.

Reply:
The simplest thing I think I can do is to change the four lines above to
```
year = np.array(stored['year']).astype(int)
```
etc, to make sure these are passed through as integers here since they may be strings if something has gone wrong.  It's not a test, but it would clarify the error messages if something is not right.  This allows the routines in `_files` to work generally while keeping things more strict at this specific use case.

Reply:
I added a suggestion below.

Reply:
I'll go with yours.  I think it is more clear if something goes wrong.